### PR TITLE
Move requestId validation to _wsOnMessage increasing test coverage by +0.4%

### PR DESCRIPTION
**Pull request by lucasrmendonca**
### Description, Motivation and Context
Move requestId validation to _wsOnMessage, removing duplicate code from multiple functions, improving test coverage and fixing 1 code climate suggestion

### What kind of change does this PR introduce?
- [x] Refactoring (no new functionality, only code improvements)

### Checklist:
- [x] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.
- [x] Overall test coverage is not decreased.
- [x] All new and existing tests passed.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/wampy.js [hunk 1]
```diff
@@ -903,7 +903,35 @@ class Wampy {
...
 
-        await handler();
+        if (this._isRequestIdValid(data)) {
+            await handler();
+        }
+    }
+
+    /**
+     * Validates the requestId for message types that need this kind of validation
+     * @param {string[]} data - [messageType, requestId]
```
Comment (KSDaemon) [inline src/wampy.js:913]: datatype is not correct here

File: src/wampy.js [hunk 2]
```diff
@@ -903,7 +903,35 @@ class Wampy {
...
+            WAMP_MSG_SPEC.UNSUBSCRIBED,
+            WAMP_MSG_SPEC.PUBLISHED,
+            WAMP_MSG_SPEC.RESULT,
+            WAMP_MSG_SPEC.REGISTERED,
+            WAMP_MSG_SPEC.UNREGISTERED
+        ].includes(messageType);
+        const isResultMessage = messageType === WAMP_MSG_SPEC.RESULT;
+
+        if (isRequestIdValidationNeeded
+            && (isResultMessage && !this._calls[requestId]
```
Comment (KSDaemon) [inline src/wampy.js:929]: Really hard to read these checks. At least they should be grouped with `()`

Comment (lucasrmendonca) [reply src/wampy.js]: Decided to split them into three different IF blocks for readability

**Review by KSDaemon — APPROVED** [review]
👍🏻 Way better and easy to read!

**Pull request merged**